### PR TITLE
refactor: Augment organization types in organization lib

### LINKED ISSUE #8499: Move B2B models from core to feature lib
Our very core model like `Order` were lately (in last 2 months) polluted with b2b-related properties (i.e. order.costCenter) . This prevents us from moving b2b-related types (i.e. CostCenter) from core to the feature lib, because core would depend then on the feature lib (Order would depend on CostCenter).

We see a hope in the ongoing ticket for _Extensible types_ (aka _Types augmentation_) https://github.com/SAP/spartacus/issues/7940

This will allow for keeping thin `Order` in the core lib, but in the my-account lib import this TS interface and _re-declare_ it with adding new properties (like `costCenter: CostCenter`). Then the core lib  won't depend on the my-account lib.

It's a follow up of https://github.com/SAP/spartacus/issues/8423
~Blocked by https://github.com/SAP/spartacus/issues/7940~

### LINKED ISSUE #8876: Check if it is worth to augment B2B models in my-account lib and simplify them in the core
For B2B checkout we have organization models in the core. Check if we need all properties and if we could use augmentation for this case.

### PR DESCRIPTION
**Pull request by marlass**
Organization models are now simpler in core library and augmented in the organization lib, where we need more properties on them.

Closes #8499 
Closes #8876

### CODE REVIEW

**Comment by marlass** [comment]
QA steps:

- augment `CostCenter` in the shell app with new property
- try to create new object of this type in app (use properties from core declaration, my-account declaration and from the augmentation in the app) and then check TS compilation (build) and autocomplete (compilation should not throw anything and editor should recommend this property on this object.
- try to use properties from my-account augmentation in core (compiler on core build should throw that the property is not defined on this model)

**Review by dunqan — DISMISSED** [review]

File: feature-libs/organization/administration/core/model/augmented-core.model.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    orgUnit?: B2BUnit;
+    selected?: boolean;
+    isAssignedToApprovers?: boolean;
+    email?: string;
+  }
+
+  interface CostCenter {
+    activeFlag?: boolean;
+    currency?: Currency;
+    originalCode?: string; // TODO: validate the need of this
```
Comment (Platonn) [inline feature-libs/organization/administration/core/model/augmented-core.model.ts]: I guess this property is needed if we copied it here. Can we remove the comment?

File: feature-libs/organization/administration/core/model/augmented-core.model.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { Currency } from '@spartacus/core';
+import { B2BApprovalProcess } from './order-approval.model';
+
+declare module '@spartacus/core' {
```
Comment (Platonn) [inline feature-libs/organization/administration/core/model/augmented-core.model.ts:3]: Can we also add here the core `Order` with it's property `permissionResults` well typed as `OrderApprovalPermissionResult[]`?
https://github.com/SAP/spartacus/blob/bc69ae2/projects/core/src/model/order.model.ts#L150

File: feature-libs/organization/administration/core/model/augmented-core.model.ts (hunk 1 above)
Comment (plabadie) [reply feature-libs/organization/administration/core/model/augmented-core.model.ts]: Done.

File: feature-libs/organization/administration/core/model/augmented-core.model.ts (hunk 2 above)
Comment (plabadie) [reply feature-libs/organization/administration/core/model/augmented-core.model.ts:3]: Good catch.  Done.

**Review by Platonn — DISMISSED** [review]

**Review by Platonn — APPROVED** [review]

**Pull request merged**